eng/scripts/Initialize-VcpkgRelease.ps1

```diff
@@ -38,29 +31,24 @@ param (
 # If there's nothing in the "port" folder to upload set SkipVcpkgUpdate to true
 # and exit. Other steps will check SkipVcpkgUpdate to decide whether to move
 # forward.
-if (!(Get-ChildItem -Path "$SourceDirectory/port")) {
+if (!(Get-ChildItem -Path "$SourceDirectory/port/CONTROL")) {
     Write-Host "###vso[task.setvariable variable=SkipVcpkgUpdate]true"
     exit
 }
 
 $packageSpec = Get-Content -Raw -Path $PackageSpecPath | ConvertFrom-Json
-$assetFilename = "$($packageSpec.packageName).tar.gz"
+$tarGzUri = "https://github.com/$GitHubRepo/archive/$($packageSpec.packageName).tar.gz" 
+
+Write-Host "Downloading tarball to compute hash" 
+$localTarGzPath = New-TemporaryFile
+Invoke-WebRequest -Uri $tarGzUri -OutFile $localTarGzPath
 
-# Upload the asset to the release
-Write-Verbose "Uploading asset: $assetFilename..."
-$assetInfo = & $PSScriptRoot/../common/scripts/New-ReleaseAsset.ps1 `
-    -ReleaseTag $packageSpec.packageName `
-    -AssetPath $SourceDirectory/$assetFilename `
-    -GitHubRepo $GitHubRepo `
-    -GitHubPat $GitHubPat
+$sha512 = (Get-FileHash -Path $localTarGzPath -Algorithm SHA512).Hash.ToLower()
+Write-Host "SHA512: $sha512"
 
-$sha512 = (Get-FileHash -Path "$SourceDirectory/$assetFilename" -Algorithm SHA512).Hash
+Write-Verbose "Writing the SHA512 hash"
+$portfileLocation = "$SourceDirectory/port/portfile.cmake"
+$newContent = Get-Content -Raw -Path $portfileLocation `
+    | ForEach-Object { $_.Replace('%SHA512%', $sha512) }
```

Comment:
If I understand correctly, when you build locally, in the file it is going to produce in the output directory,  `%SHA512%` is going to be replaced with just an empty string. If so, since we are updating this thing via regex and not via cmake, can we actually look for `SHA512\s+1\s.$` ("SHA512", at least one space, "1", any number of spaces, end of line)?

In this case, the files that it produces in the local `/out` directory are still usable for local testing and experimentation: you copy them to your local copy of vcpkg, it does not fail on parsing the portfile.cmake file, and gives you the SHA it expects to see. In my opinion, it is a convenience for developers, and especially for non-MS contributors, who don't see the full extent of things that are going in our CI/CD. Plus, an error that SHA does not match is easier to understand, than that parse failed.  Putting "SHA512 1" is what VcPkg recomend in their developer docs.